webcompat/webhooks/model.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""WebCompat Issue Model for webhooks."""
+
+from dataclasses import dataclass, field, InitVar
+from typing import Any, Dict, List
+
+from webcompat import app
+from webcompat.webhooks.helpers import extract_metadata
+from webcompat.webhooks.helpers import get_issue_labels
+from webcompat.webhooks.helpers import make_request
+from webcompat.webhooks.helpers import prepare_rejected_issue
+
+PUBLIC_REPO = app.config['ISSUES_REPO_URI']
+PRIVATE_REPO = app.config['PRIVATE_REPO_URI']
+
+
+@dataclass
+class WebHookIssue:
+    """WebCompat Issue Model for WebHook consumption"""
+    payload: InitVar[Dict[str, Any]]
```

Comment:
This is a pretty lazy type annotation for (parsed) JSON, we're trusting that GitHub doesn't send us scary thing.

Reply:
@rik proposed something else for initialization.

https://gist.github.com/rik/d7d195ee5cc11b16d0d951312905eee3#gistcomment-3396628

which is nice too with less code.

Reply:
(for context)
he proposed it because I talked about `InitVar` on 
https://www.la-grange.net/2020/07/29/plantez

as I was discovering them during the review today.

Reply:
I see, so the idea is to add a static `from_dict`, which will do the initialization.

I like this solution, it seems less confusing to read (especially for newcomers to dataclasses like me) -- also it means we can ditch the `field(init=False)` stuff I think.

Reply:
> also it means we can ditch the `field(init=False)` stuff I think

(no, we can't, because we need to still do the data manipulation in `__post_init__()`)

Reply:
> (no, we can't, because we need to still do the data manipulation in `__post_init__()`)

I'm a little slow... we can do the data manipulation in the classmethod. :) So, no need for field(init=False). :))